### ISSUE TITLE
Enhancement: detect condition on exit code while set -e

### ISSUE THREAD

**Issue by secnum** [koalaman/shellcheck#1372]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
set -o errexit
id -u test >/dev/null 2>&1
if [[ "${?}" -eq 1 ]]; then
	sudo useradd test
fi
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Possible unwanted stopping of the script because a test on a command exit code is performed while set -e variable is used.
